redbot/cogs/reports/reports.py

```diff
@@ -334,7 +334,15 @@ async def on_message(self, message: discord.Message):
                 self.tunnel_store[k]["msgs"] = msgs
 
         for key in to_remove:
-            self.tunnel_store.pop(key, None)
+            if tun := self.tunnel_store.pop(key, None):
```

Comment:
This actually assigns a dictionary to `tun` - the actual Tunnel object seems to be in `tun["tun"]`

Reply:
Yep, you're right, 1 sec.